sopel/privileges.py

```diff
@@ -1,146 +1,172 @@
-"""Constants for user privileges in channels.
+"""Constants for user privileges in channels."""
+# Copyright 2023 dgw, technobabbl.es
+from __future__ import annotations
 
-Privilege levels
-================
+import enum
 
-Historically, there were two user privileges in channels:
 
-* :data:`OP`: channel operator, or chanop, set and unset by ``+o`` and ``-o``
-* :data:`VOICE`: the privilege to send messages to a channel with the
-  ``+m`` mode, set and unset by ``+v`` and ``-v``
+__all__ = [
+    'AccessLevel',
+]
 
-Since then, other privileges have been adopted by IRC servers and clients:
 
-* :data:`HALFOP`: intermediate level between Voiced and OP, set and unset by
-  ``+h`` and ``-h``
-* :data:`ADMIN`: channel admin, above OP and below OWNER, set and unset by
-  ``+a`` and ``-a``
-* :data:`OWNER`: channel owner, above ADMIN and OP, set and unset by ``+q`` and
-  ``-q``
+class AccessLevel(enum.IntFlag):
+    """Enumeration of available user privilege levels.
 
-.. important::
+    Privilege levels
+    ================
 
-    Not all IRC networks support these added privilege modes. If you are
-    writing a plugin for public distribution, ensure your code behaves sensibly
-    if only +v (voice) and +o (op) modes exist.
+    Historically, there were two user privilege levels in a channel:
 
-Compare privileges
-==================
+    * :data:`~AccessLevel.OP`: channel operator, set and unset by modes ``+o``
```

Comment:
I haven't tested this, but maybe this would work as well?

```suggestion
    * :attr:`~AccessLevel.OP`: channel operator, set and unset by modes ``+o``
```